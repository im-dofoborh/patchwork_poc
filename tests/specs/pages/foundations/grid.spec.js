import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import GridPage from '~/pages/foundations/grid';
import { site } from '~/tests/fixtures';

describe('Grid page', () => {
  let localVue = null;
  let wrapper = null;
  let store = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('grid-page', GridPage);
    localVue.use(Vuex);
    store = new Vuex.Store({
      state: {
        currentSite: site,
      },
    });
    wrapper = mount(GridPage, {
      localVue,
      store,
    });
  })

  it('renders', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
