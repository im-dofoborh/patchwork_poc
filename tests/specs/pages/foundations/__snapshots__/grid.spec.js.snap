// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid page renders 1`] = `
<div class="sg-content container">
  <div class="sg-content__header">
    <h1 class="heading-1">Grid</h1>
  </div>
  <div class="sg-content__section editor-content">
    <h2>Overview</h2>
    <p>
      Our grid system uses Bootstrap 4 Alpha 6 with a custom configuration. See the Bootstrap
      <a href="https://getbootstrap.com/docs/4.1/layout/overview/">grid documentation</a>
      to understand how to use the grid.
    </p>
    <p>
      All pages within Slate should use an identical grid. This is partly
      down to our usage of ad skins which requires a consistency in our grid between pages. See our
      <a href="/patchwork/adverts">adverts</a> documentation or
      <a href="https://immediateco.atlassian.net/wiki/spaces/WCPP/pages/5181172/Grid#Grid-Changestothegridcausedbyadskins(otherwiseknownastakeoversorroadblocks)">
        Confluence
      </a>
      for more information.
    </p>
  </div>
  <div class="sg-content__section">
    <div class="editor-content">
      <h3 class="heading-3">Extra Small (Mobile)</h3>
      <p>
        <!---->
        we can use up to 12 columns in any combination within the grid with
        a maximum width of 100% and a 20px gutter between columns.
        This results in a minimum outer gutter of 10px.
      </p>
    </div>
    <div class="sg-mock-grid container" style="width: 320px;">
      <div class="sg-mock-grid__row row">
        <div class="col-12">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-6">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-6">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
    </div>
  </div>
  <div class="sg-content__section">
    <div class="editor-content">
      <h3 class="heading-3">Small (Phablet)</h3>
      <p>
        Up to 767px

        we can use up to 12 columns in any combination within the grid with
        a maximum width of 576px and a 20px gutter between columns.
        This results in a minimum outer gutter of 10px.
      </p>
    </div>
    <div class="sg-mock-grid container" style="width: 576px;">
      <div class="sg-mock-grid__row row">
        <div class="col-12">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-6">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-6">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
    </div>
  </div>
  <div class="sg-content__section">
    <div class="editor-content">
      <h3 class="heading-3">Medium (Tablet)</h3>
      <p>
        Up to 991px

        we can use up to 12 columns in any combination within the grid with
        a maximum width of 748px and a 20px gutter between columns.
        This results in a minimum outer gutter of 20px.
      </p>
    </div>
    <div class="sg-mock-grid container" style="width: 748px;">
      <div class="sg-mock-grid__row row">
        <div class="col-12">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-6">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-6">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
    </div>
  </div>
  <div class="sg-content__section">
    <div class="editor-content">
      <h3 class="heading-3">Large (Desktop)</h3>
      <p>
        From 992px

        we can use up to 12 columns in any combination within the grid with
        a maximum width of 960px and a 20px gutter between columns.
        This results in a minimum outer gutter of 26px.
      </p>
    </div>
    <div class="sg-mock-grid container" style="width: 960px;">
      <div class="sg-mock-grid__row row">
        <div class="col-12">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-6">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-6">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-4">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-3">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-2">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
      <div class="sg-mock-grid__row row">
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
        <div class="col-1">
          <div class="sg-mock-grid__column"></div>
        </div>
      </div>
    </div>
  </div>
  <div class="sg-content__section editor-content">
    <h2>Fixed columns</h2>
    <p>
      Due to having to support fixed-width adverts, it is sometimes necessary to ensure that
      within a layout, a single column is set to a width of 300px. The other column will use
      the remaining space and behave like a typical Bootstrap grid.
    </p>
    <p>
      We can achieve this by using the following layout below. Details on how to implement
      this are available in
      <a href="https://immediateco.atlassian.net/wiki/spaces/WCPP/pages/5181172/Grid#Grid-Achievingfixedwidthcolumns">Confluence</a>.
    </p>
    <p>
      'When the 'small' and 'extra small' breakpoints are hit, these columns will follow typical
      bootstrap column behaviour and will fit 100% width and stack.
    </p>
  </div>
  <div class="sg-content__section">
    <div class="editor-content">
      <h3 class="heading-3">Medium (Tablet)</h3>
    </div>
    <div class="sg-mock-grid container" style="width: 748px;">
      <div class="sg-mock-grid__row row">
        <div class="sg-mock-grid__rail-container">
          <div class="sg-mock-grid__rail sg-mock-grid__rail--primary">
            <div class="sg-mock-grid__column">
              Fluid bootstrap grid
            </div>
          </div>
          <div class="sg-mock-grid__rail sg-mock-grid__rail--secondary">
            <div class="sg-mock-grid__column sg-mock-grid__column--rail">
              Fixed to 300px
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="sg-content__section">
    <div class="editor-content">
      <h3 class="heading-3">Large (Desktop)</h3>
    </div>
    <div class="sg-mock-grid container" style="width: 960px;">
      <div class="sg-mock-grid__row row">
        <div class="sg-mock-grid__rail-container">
          <div class="sg-mock-grid__rail sg-mock-grid__rail--primary">
            <div class="sg-mock-grid__column">
              Fluid bootstrap grid
            </div>
          </div>
          <div class="sg-mock-grid__rail sg-mock-grid__rail--secondary">
            <div class="sg-mock-grid__column sg-mock-grid__column--rail">
              Fixed to 300px
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
